mam/examples/ntru-key-exchange.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https:github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ * [ITSec Lab]
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include <stdio.h>
+#include "mam/api/api.h"
+#include "mam/ntru/ntru.h"
+
+int main() {
+  //Generating a keypair for the receiver api
+  mam_prng_t prng;
+  mam_ntru_sk_t ntru;
+  MAM_TRITS_DEF(nonce, 3 * 10);
+  MAM_TRITS_DEF(key, MAM_PRNG_SECRET_KEY_SIZE);
+  nonce = MAM_TRITS_INIT(nonce, 3 * 10);
+  key = MAM_TRITS_INIT(key, MAM_PRNG_SECRET_KEY_SIZE);
+  trits_from_str(key,
+                 "AAABBBCCCAAABBBCCCAAABBBCCC"
+                 "AAABBBCCCAAABBBCCCAAABBBCCC"
+                 "AAABBBCCCAAABBBCCCAAABBBCCC");
+  mam_prng_init(&prng, key);
+  ntru_sk_reset(&ntru);
+  ntru_sk_gen(&ntru, &prng, nonce);
+  //Setting up two APIs
+  mam_api_t sender_api;
+  tryte_t *sender_seed = (tryte_t *)"SENDERSEED9SENDERSEED9SENDERSEED9SENDERSEED9SENDERSEED9SENDERSEED9SENDERSEED99999";
+  mam_api_init(&sender_api, sender_seed);
+  mam_api_t receiver_api;
+  tryte_t *receiver_seed = (tryte_t *)"RECEIVERSEED9RECEIVERSEED9RECEIVERSEED9RECEIVERSEED9RECEIVERSEED9RECEIVERSEED9999";
+  mam_api_init(&receiver_api, receiver_seed);
+  //Setting the generated keypair to the receiver API
+  mam_api_add_ntru_sk(&receiver_api, &ntru);
```

Comment:
error check (applies for all none void calls here)
maybe use `ERR_BIND_RETURN` or `ERR_BIND_GOTO`

Reply:
Do you have an example for me?

Reply:
found one. done.